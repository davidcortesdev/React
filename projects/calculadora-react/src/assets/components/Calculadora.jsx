import { useState } from "react"

export function Calculadora() {

    const [value, setValue] = useState('')

    function handleValue (event){

        console.log(`Valor: ${value}`)
        console.log(`Evento: ${event.target.value}`)
        const newValue = event.target.value
        const lastValue = value.toString().charAt(value.length - 1)
        
        console.log(`Ã™ltimo valor: ${lastValue}`)

        if (['*', '-', '/', '+','='].includes(lastValue) && ['*', '-', '/', '+','='].includes(newValue)) {
            return
        }

        
        setValue((prevValue) => prevValue + newValue)
        console.log('click')

        
    }

    function calculateTotal(){
        setValue(eval(value))
    }

    function limpiar(){
        setValue('')
    }

    return(
        <section className='seccion-calculadora'>
        <input type='text' value={value} className='input-calculadora' readOnly></input>
        <div className='container-calculadora'>
            <button className='boton-calculadora' value='7' onClick={handleValue}>7</button>
            <button className='boton-calculadora' value='8' onClick={handleValue}>8</button>
            <button className='boton-calculadora' value='9' onClick={handleValue}>9</button>
            <button className='boton-calculadora' value='/' onClick={handleValue}>/</button>
            <button className='boton-calculadora' value='4' onClick={handleValue}>4</button>
            <button className='boton-calculadora' value='5' onClick={handleValue}>5</button>
            <button className='boton-calculadora' value='6' onClick={handleValue}>6</button>
            <button className='boton-calculadora' value='*' onClick={handleValue}>X</button>
            <button className='boton-calculadora' value='1' onClick={handleValue}>1</button>
            <button className='boton-calculadora' value='2' onClick={handleValue}>2</button>
            <button className='boton-calculadora' value='3' onClick={handleValue}>3</button>
            <button className='boton-calculadora' value='-' onClick={handleValue}>-</button>
            <button className='boton-calculadora' value='0' onClick={handleValue}>0</button>
            <button className='boton-calculadora' value='.' onClick={handleValue}>.</button>
            <button className='boton-calculadora' value='=' onClick={calculateTotal}>=</button>
            <button className='boton-calculadora' value='+' onClick={handleValue}>+</button>
        </div>
            <button className='boton-clean' onClick={limpiar}>Limpiar</button>
      </section>
    )
}