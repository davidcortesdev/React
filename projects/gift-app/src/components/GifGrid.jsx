import PropTypes from 'prop-types'
import { GifItem } from './GifItem'
import { useLoadGifs } from '../hooks/useLoadGifs'

export const GifGrid = ({ category }) => {


    const { gifs, isLoading } = useLoadGifs( category )
    
    return(
        <>
            <h3>{ category }</h3>
            {
                isLoading && <h2>Cargando...</h2>
            }
            <div className='card-grid'>
            {
                gifs.map( ( gif ) => (
                    <GifItem key={gif.id} {...gif}/>
                ))
            }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}