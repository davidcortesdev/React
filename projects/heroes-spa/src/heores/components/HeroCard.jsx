import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

export const HeroCard = ({ id, superheroe, alter_ego, first_appearance, characters }) => {

    const heroImageUrl = `/assets/heroes/${ id }.jpg`
    
    const CharactersByHero = ({characters, alter_ego}) => {
        return (alter_ego === characters)
            ? <></>
            : <p> { characters } </p>

    }

  return (
    <div className="col animate__animated animate__fadeIn">
        <div className="card">
            <div className='row no-gutter'>
                <div className='col-4'>
                    <img src={heroImageUrl} className='card-img' alt={ superheroe }/>
                </div>

                <div className="col-8">
                    <div className="card-body">
                        
                        <h5 className='card-title'> {superheroe} </h5>
                        
                        <p className='card-text'> {alter_ego} </p>
                        
                        <CharactersByHero characters={characters} alter_ego={alter_ego}/>
                        
                        <p className='card-text'>
                            <small className='text-muted'> {first_appearance} </small>
                        </p>

                        <Link to={`/hero/${ id }`}>
                            Mas...
                        </Link>
                        
                    </div>
                </div>
                
            </div>
        </div>
    </div>
  )
}
{            /* <h1>{ superheroe }</h1>
                <div>Publisher: { publisher } </div>
                <div>Alter Ego: { alter_ego } </div>
                <div>First Appearance: { first_appearance } </div>
                <div>Characters: { characters } </div> */}
HeroCard.propTypes = {
    id: PropTypes.string,
    superheroe: PropTypes.string,
    publisher: PropTypes.string,
    alter_ego: PropTypes.string,
    first_appearance: PropTypes.string,
    characters: PropTypes.string,
}