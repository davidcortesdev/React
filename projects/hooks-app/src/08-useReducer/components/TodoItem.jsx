import PropTypes from 'prop-types'

export const TodoItem = ({ todo = {}, onDeleteTodo, onToggleTodo }) => {

    
  
    return (
        <li className={`list-group-item d-flex justify-content-between ${todo.done ? 'bg-danger text-light' : ''}`}>
            <span 
                className={`align-self-center ${todo.done ? 'text-decoration-line-through' : ''}`}
                onClick={ () => onToggleTodo( todo.id )}
                aria-label='span'> 
                
                {todo.description} 
            </span>

            <button className={`btn btn-danger ${todo.done ? 'btn-light' : ''}`} 
                    onClick={ () => onDeleteTodo(todo.id)}> 
                Borrar 
            </button>
        </li>
  )
}

TodoItem.propTypes = {
    todo: PropTypes.object,
    onDeleteTodo: PropTypes.func,
    onToggleTodo: PropTypes.func
}