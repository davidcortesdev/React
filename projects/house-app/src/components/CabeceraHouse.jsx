import '../css/CabeceraHouse.css'

export const CabeceraHouse = () => {
  return (
    <header className="header-house">
        <a className="enl-inicio" href=""><h1>Organiza tu casa</h1></a>
    </header>
  )
}
