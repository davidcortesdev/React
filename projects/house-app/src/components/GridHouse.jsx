import '../css/GridHouse.css'
import kitchen from '../../public/cocina.png'
import bathroom from '../../public/baño.png'
import livingroom from '../../public/comedor.png'
import room from '../../public/habitacion.png'

export const GridHouse = () => {
  return (
    <section className="section-grid-house">

      <div className='grid-house'>

        <div className='tarjeta1-house'>
          <img className="kitchen" src={kitchen} alt="" />
          <h2 className='tarjeta1-text-house'>Cocina</h2>
        </div>

        <div className='tarjeta2-house'>
          <img className="bathroom" src={bathroom} alt="" />
          <h2 className='tarjeta2-text-house'>Baño</h2>
        </div>
          
        
        <div className='tarjeta3-house'>
          <img className="room" src={room} alt="" />
          <h2 className='tarjeta3-text-house'>Habitación</h2>
        </div>

        <div className='tarjeta4-house'>
          <img className="livingroom" src={livingroom} alt="" />
          <h2 className='tarjeta4-text-house'>Comedor</h2>
        </div>
        
      </div>

    </section>
  )
}
