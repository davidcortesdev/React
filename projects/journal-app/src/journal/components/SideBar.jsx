import { Box, Divider, Drawer, List, Toolbar, Typography } from "@mui/material"
import { useSelector } from "react-redux"
import { SideBarItem } from "./"
import PropTypes from 'prop-types'

export const SideBar = ({ drawerWidth }) => {

    const { displayName } = useSelector( state => state.auth)
    const { notes } = useSelector( state => state.journal)

  return (
    
    <Box
        component='nav'
        sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >

        <Drawer
            variant="permanent"
            open
            sx={{ 
                display: { xs: 'block' },
                '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
             }}
        >

            <Toolbar>
                <Typography variant="h6" noWrap component='div'>
                    { displayName }
                </Typography>
            </Toolbar>

            <Divider />

            <List>
                {
                    notes.map( nota => (
                        <SideBarItem key={ nota.id } {...nota} />
                    ))
                }
            </List>

        </Drawer>

    </Box>
  )
}


SideBar.propTypes = {
    drawerWidth: PropTypes.number
}