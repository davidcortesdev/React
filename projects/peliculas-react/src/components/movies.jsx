/* eslint-disable react/prop-types */
function ListOfMovies ({ movies }) {
 return (
    <ul className='lista-peliculas'>
     {
        movies.map(movie => (
        <li className='pelicula' key={movie.id}>
            <h1>{movie.title}</h1>
            <p>{movie.year}</p>
            <img src={movie.image} alt={movie.title} />
        </li>
        ))
     }
    </ul>
 )
}

function NoMoviesResults () {
   return(
    <p>No se encontraron películas para esta búsqueda</p>
    )
}

export function Movies ({ movies }) {
  const hasMovies = movies?.length > 0

    return(
        hasMovies
            ? <ListOfMovies movies={movies}/>
            : <NoMoviesResults />
    )
}