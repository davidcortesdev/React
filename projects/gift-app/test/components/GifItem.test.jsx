import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

/* eslint-disable no-undef */
describe('Pruebas en <GifItem />', () => { 

    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg'

    test('Match con el snapshot de <GifItem />', () => { 
        
        const {container} = render(<GifItem title={title} url={url}/>)
        expect( container ).toMatchSnapshot()

     })

     test('Debe mostrar la imagen con el URL y el ALT indicado', () => { 
        
        render(<GifItem title={ title } url={ url }/>)
        //screen.debug()
        const {src, alt} = screen.getByRole('img') // -> screen.getByRole('img').src
        expect( src ).toBe( url )
        expect( alt ).toBe( title )

      })


      test('Debe mostrar el titulo en el componente', () => { 
        
        render(<GifItem title={ title } url={ url }/>)
        expect( screen.getByText( title )).toBeTruthy()

       })

 })