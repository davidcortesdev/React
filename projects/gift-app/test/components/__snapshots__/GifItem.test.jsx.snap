// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Match con el snapshot de <GifItem /> 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
